moveit_simple/src/moveit_simple.cpp

```diff
@@ -775,6 +775,44 @@ bool Robot::getIK(const Eigen::Affine3d pose, std::vector<double> & joint_point,
 
 
 
+bool Robot::isNearSingular(const std::vector<double> & joint_point) const
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+  double threshold = 1e-10; // How to determine this threshold?
+  std::vector<double> local_joint_point = joint_point;
+
+  // reference point for Jacobian
+  Eigen::Vector3d reference_point(0,0,0);
+  if (joint_point.empty())
+  {
+    ROS_DEBUG_STREAM("Empty joint point passed to isNearSingular, using current state");
+//    local_joint_point = getJointState(); // Update after PR#21
+    robot_state_->copyJointGroupPositions(joint_group_->getName(), local_joint_point);
+  }
+  robot_state_->setJointGroupPositions(joint_group_, local_joint_point);
+
+  Eigen::MatrixXd jacobian;
+  if (robot_state_->getJacobian(joint_group_, joint_group_->getLinkModels().back(),
+                                  reference_point, jacobian, false))
+  {
+    // Check for determinant of J*J'
+    if (fabs((jacobian*jacobian.transpose()).determinant()) <= threshold)
+      {
+        ROS_WARN_STREAM("Given configuration " <<local_joint_point <<
+                                                        " is near Singularity");
+        return true;
+      }else{
+        ROS_INFO_STREAM("Given configuration " <<local_joint_point <<
+                                               " is away from any Singularity");
+        return false;
+      }
+  }else{
+    ROS_ERROR_STREAM("Jacobian not found for " << local_joint_point);
+    throw std::invalid_argument("joint_group " + joint_group_->getName() +
```

Comment:
Go ahead and log the error and just return false.  Let's not worry about what may be in the future.